server/__init__.py

```diff
@@ -167,6 +167,79 @@ def challengeSaved(event):
         challenge, folder, save=True)
 
 
+def onSubmissionCreated(event):
+    """
+    Hook into submission created event to set up access control on the
+    submission folder. Phase admins should have read access.
+    """
+    submission = event.info
+
+    folderModel = ModelImporter.model('folder')
+    userModel = ModelImporter.model('user')
+    phaseModel = ModelImporter.model('phase', 'challenge')
+
+    # Get phase admin user ids
+    folder = folderModel.load(submission['folderId'], force=True, exc=True)
+    phase = phaseModel.load(submission['phaseId'], force=True, exc=True)
+    phaseAcl = phaseModel.getFullAccessList(phase)
+    phaseAdminUsers = [userModel.load(user['id'], force=True, exc=True)
+                       for user in phaseAcl.get('users')
+                       if (user['level'] >= AccessType.WRITE and
+                           user['id'] != folder['creatorId'])]
+
+    # Update phase submission folder ACL for phase admins
+    for user in phaseAdminUsers:
+        folderModel.setUserAccess(folder, user, AccessType.READ)
+    if phaseAdminUsers:
+        folderModel.save(folder, validate=False)
```

Comment:
My one suggestion would be to move this logic out of the event hook and into the submission model's `createSubmission` event directly, since it's part of the same plugin.